reactor-pool/src/test/java/reactor/pool/CommonPoolTest.java

```diff
@@ -2680,4 +2681,50 @@ void testIssue_174(PoolStyle style) {
 		assertThat(pool.config().allocationStrategy().estimatePermitCount()).isEqualTo(0);
 		assertThat(pool.metrics().idleSize()).isEqualTo(10);
 	}
+
+	@ParameterizedTestWithName
+	@MethodSource("allPools")
+	@Tag("metrics")
+	void recordsPendingCountAndLatencies(PoolStyle configAdjuster) {
+		PoolBuilder<String, ?> builder = PoolBuilder
+				.from(Mono.defer(() -> Mono.just("foo")))
+				.metricsRecorder(recorder)
+				.sizeBetween(0, 1)
+				.clock(recorder.getClock());
+		Pool<String> pool = configAdjuster.apply(builder);
+
+		PooledRef<String> pooledRef = pool.acquire(Duration.ofMillis(1)).block(Duration.ofSeconds(1)); //success
+		assertThat(pooledRef).isNotNull();
+
+		pool.acquire(Duration.ofMillis(50)).subscribe(); //success
+
+		pool.acquire(Duration.ofMillis(1))
+				.as(StepVerifier::create)
+				.expectError(PoolAcquireTimeoutException.class)
+				.verify(Duration.ofSeconds(1)); //error
+
+		pooledRef.release().block(Duration.ofSeconds(1));
+
+		assertThat(recorder.getPendingTotalCount())
+				.as("total pending")
+				.isEqualTo(2);
+		assertThat(recorder.getPendingSuccessCount())
+				.as("pending success")
+				.isEqualTo(1)
```

Comment:
The above comments say "success" twice, perhaps it makes sense to clarify a bit which one triggers the successful pending increment.

Reply:
OK I'll add to the comment that the second one is pending. Thanks.

Reply:
comments extended